app/service/database_dump.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import re
+import subprocess
+from datetime import datetime
+from pathlib import Path
+
+from app.config import (
+    ADMIN_POSTGRES_DATABASE,
+    ADMIN_POSTGRES_HOST,
+    ADMIN_POSTGRES_PASSWORD,
+    ADMIN_POSTGRES_USER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+
+def validate_postgres_param(value: str) -> str:
+    """
+    Allow only alphanumeric, dots, underscores and hyphens in Postgres parameters.
+
+    :param value: The Postgres parameter to validate
+    returns str: The validated parameter
+    :raises ValueError: If the parameter is invalid
+    """
+    if value is None or not re.match(r"^[\w.-]+$", value):
+        raise ValueError(f"Invalid Postgres parameter: {value}")
+    return value
+
+
+def delete_local_file(file_path: str) -> None:
+    """Delete a local file safely, logging any issues."""
+    path = Path(file_path)
+
+    try:
+        if path.exists():
+            path.unlink()
+            _LOGGER.debug(f"Deleted local file: {file_path}")
+    except Exception as e:
+        _LOGGER.error(f"⚠️ Failed to delete file {file_path}: {e}")
+        raise
+
+
+def get_database_dump() -> str:
```

Comment:
suggestion: make `timeout` an optional parameter here with a default of 300.
```suggestion
def get_database_dump(timeout_secs: int = 300) -> str:
```